comp/core/workloadmeta/collectors/internal/ecs/ecs.go

```diff
@@ -49,12 +54,19 @@ type resourceTags struct {
 
 // NewCollector returns a new ecs collector provider and an error
 func NewCollector() (workloadmeta.CollectorProvider, error) {
+	orchestratorECSCollectionEnabled := false
+	if config.Datadog.GetBool("orchestrator_explorer.enabled") &&
+		config.Datadog.GetBool("orchestrator_explorer.ecs_collection.enabled") {
+		orchestratorECSCollectionEnabled = true
+	}
+
 	return workloadmeta.CollectorProvider{
 		Collector: &collector{
-			id:           collectorID,
-			resourceTags: make(map[string]resourceTags),
-			seen:         make(map[workloadmeta.EntityID]struct{}),
-			catalog:      workloadmeta.NodeAgent | workloadmeta.ProcessAgent,
+			id:                               collectorID,
+			resourceTags:                     make(map[string]resourceTags),
+			seen:                             make(map[workloadmeta.EntityID]struct{}),
+			catalog:                          workloadmeta.NodeAgent | workloadmeta.ProcessAgent,
+			orchestratorECSCollectionEnabled: orchestratorECSCollectionEnabled,
```

Comment:
Should it be enabled only in the process-agent ?

Reply:
It won't run in process-agent because the new check will only run in core agent.
Thanks, it is now fixed
https://github.com/DataDog/datadog-agent/blob/f67c3d667e00dcf6350a9d80a7075998e57ad9e3/comp/core/workloadmeta/collectors/internal/ecs/ecs.go#L69